src/go/rpk/pkg/cli/cmd/root.go

```diff
@@ -89,3 +104,85 @@ func Execute() {
 		os.Exit(1)
 	}
 }
+
+type pluginHandler interface {
+	lookPath(file string) (path string, ok bool)
+	exec(path string, args []string) error
+}
+
+// tryExecPlugin looks for a plugin, following the exact behavior of kubernetes:
+//
+//  - all dashes are turned to underscores
+//  - all "pieces" (non-flags) are joined with a dash
+//  - we prefer the longest command match
+//  - we search upward by piece until we run out of pieces
+//  - lastly, the command must be executable
+//
+// Dashes are only replaced into underscores during the path search; if an
+// argument hash a dash and is used as an argument to a plugin, the dash
+// remains.
+//
+// So,
+//
+//     rpk foo-bar baz boz fizz-buzz --flag
+//
+// is translated into searching and execing (with osPluginHandler), in order:
+//
+//     rpk-foo_bar-baz-boz-fizz_buzz (with args "--flag")
+//     rpk-foo_bar-baz-boz           (with args "fizz-buzz --flag")
+//     rpk-foo_bar-baz               (with args "boz fizz-buzz --flag")
+//     rpk-foo_bar                   (with args "baz boz fizz-buzz --flag")
+//
+// If a plugin is run, this returns the run error and true, otherwise this
+// returns false.
+func tryExecPlugin(h pluginHandler, args []string) (string, error) {
+	var pieces []string
+	for _, arg := range args {
+		if strings.HasPrefix(arg, "-") { // found a flag, quit
+			break
+		}
+		pieces = append(pieces, strings.Replace(arg, "-", "_", -1))
+	}
+
+	if len(pieces) == 0 {
+		return "", nil // no plugin specified (command is just "rpk")
+	}
+
+	foundPath := ""
+	for len(pieces) > 0 {
+		path, ok := h.lookPath("rpk-" + strings.Join(pieces, "-"))
+		if !ok { // did not find with this piece, strip and search higher
+			pieces = pieces[:len(pieces)-1]
+			continue
+		}
+		foundPath = path
+		break
+	}
+	if len(foundPath) == 0 {
+		return "", nil
+	}
+
+	return foundPath, h.exec(foundPath, args[len(pieces):])
+}
+
+type osPluginHandler struct{}
+
+func (*osPluginHandler) lookPath(file string) (string, bool) {
+	path, err := exec.LookPath(file)
+	return path, err == nil
+}
+func (*osPluginHandler) exec(path string, args []string) error {
+	args = append([]string{path}, args...)
+	env := os.Environ()
+	if runtime.GOOS == "windows" {
+		return (&exec.Cmd{
```

Comment:
Is this for WSL2?

Reply:
This is just for Windows in general. I don't think redpanda is supported on Windows, but a client should be able to connect from Windows to get the status, and an exec like this can support a module on Windows as well.

Reply:
The reason I was asking is that we don't build for windows right now, so even though this is nice, it won't be used.